boot/bootutil/include/bootutil/image.h

```diff
@@ -89,6 +95,11 @@ struct flash_area;
 #define IMAGE_TLV_PUBKEY            0x02   /* public key */
 #define IMAGE_TLV_SHA256            0x10   /* SHA256 of image hdr and body */
 #define IMAGE_TLV_SHA384            0x11   /* SHA384 of image hdr and body */
+#define IMAGE_TLV_RAW_SLOT_SHA256   0x18   /*
+                                            * SHA256 of compressed/encrypted image data in the slot
+                                            * which should be checked to ensure image integrity
+                                            * before decrypting/decompressing data
```

Comment:
Isn't this little confusing? In case when image is encrypted actually of the other SHA TLV are used but the image is first decrypted and then sha is calculated on the raw image.
So if only the existing encryption is used the other TLV serve as the RAW sha.
Shouldn't we have opposite?

Currently what happens is the `bootutil_img_validate` will first call `bootutil_img_hash` that will decrypt image into RAM and calculate the sha with pre-selected, during compilation, sha algorithm and then compare the resulting sha with expected TLV (EXPECTED_HASH_TLV ), which would be the IMAGE_TLV_SHA256 by default.

This also means that if we validate slot, that has image already decrypted and decompressed we will check it against EXPECTED_HASH_TLV  stored sha, which currently serves as the "RAW" sha.